static/src/javascripts/projects/common/modules/experiments/ab-ophan.js

```diff
@@ -72,49 +60,39 @@ const buildOphanSubmitter = (
  *
  * @see {@link defersImpression}
  */
-const registerCompleteEvent = (complete: boolean) => (test: ABTest): void => {
-    const variantId = getTestVariantId(test.id);
-
-    if (variantId && variantId !== 'notintest') {
-        const variant = getVariant(test, variantId);
-
-        if (variant != null) {
-            const listener =
-                (complete ? variant.success : variant.impression) || noop;
-
-            try {
-                listener(buildOphanSubmitter(test, variantId, complete));
-            } catch (err) {
-                reportError(err, {}, false);
-            }
-        }
+const registerCompleteEvent = (complete: boolean) => (
+    test: Runnable<ABTest>
+): void => {
+    const variant = test.variantToRun;
+    const listener = (complete ? variant.success : variant.impression) || noop;
+
+    try {
+        listener(buildOphanSubmitter(test, variant, complete));
+    } catch (err) {
+        reportError(err, {}, false);
     }
 };
 
-export const registerCompleteEvents = (tests: $ReadOnlyArray<ABTest>): void =>
-    tests.forEach(registerCompleteEvent(true));
+export const registerCompleteEvents = (
+    tests: $ReadOnlyArray<Runnable<ABTest>>
+): void => tests.forEach(registerCompleteEvent(true));
 
-export const registerImpressionEvents = (tests: $ReadOnlyArray<ABTest>): void =>
-    tests.filter(defersImpression).forEach(registerCompleteEvent(false));
+export const registerImpressionEvents = (
+    tests: $ReadOnlyArray<Runnable<ABTest>>
+): void => tests.filter(defersImpression).forEach(registerCompleteEvent(false));
 
-export const buildOphanPayload = (): OphanABPayload => {
+export const buildOphanPayload = (
+    tests: $ReadOnlyArray<Runnable<ABTest>>
```

Comment:
having this accept an array of `Runnable<ABTest>>`, instead of fetching the tests itself in the body of the function, simplifies the logic of this file and means it's no longer sensitive to the order of execution.